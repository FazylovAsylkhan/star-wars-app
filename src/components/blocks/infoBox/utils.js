function getArrPlanetsData(objectData) {
  const arrInfoHomeworld = [
    `Population: ${objectData?.population}`,
    `Diameter: ${objectData?.diameter}`,
    `Rotation period: ${objectData?.rotation_period}`,
    `Climate: ${objectData?.climate}`,
    `Gravity: ${objectData?.gravity}`,
    `Terrain: ${objectData?.terrain}`,
    `Surface water: ${objectData?.surface_water}`,
    `Orbital period: ${objectData?.orbital_period}`,
  ];
  return arrInfoHomeworld;
}
function getArrStarshipsData(objectData) {
  const arrInfoHomeworld = [
    `model: ${objectData?.model}`,
    `MGLT: ${objectData?.MGLT}`,
    `Consumables: ${objectData?.consumables}`,
    `Crew: ${objectData?.crew}`,
    `Hyperdrive rating: ${objectData?.hyperdrive_rating}`,
    `Length: ${objectData?.hyperdrive_rating}`,
    `Manufacturer: ${objectData?.manufacturer}`,
    `Max atmosphering speed: ${objectData?.max_atmosphering_speed}`,
    `Passengers: ${objectData?.passengers}`,
    `Starship class: ${objectData?.starship_class}`,
  ];
  return arrInfoHomeworld;
}
function getArrPeopleData(objectData) {
  const arrInfoPerson = [
    `Gender: ${objectData?.gender}`,
    `Height: ${objectData?.height}`,
    `Mass: ${objectData?.mass}`,
    `Hair color: ${objectData?.hair_color}`,
    `Skin color: ${objectData?.skin_color}`,
    `Eye color: ${objectData?.eye_color}`,
    `Birth year: ${objectData?.birth_year}`,
  ];
  return arrInfoPerson;
}

export function getArrHomeworldData(objectData) {
  const arrInfoHomeworld = [
    `Homeworld: ${objectData?.name}`,
    `Population: ${objectData?.population}`,
    `Diameter: ${objectData?.diameter}`,
    `Rotation period: ${objectData?.rotation_period}`,
  ];
  return arrInfoHomeworld;
}

export function getRelativeUrl(url) {
  return url.split("/").splice(4);
}
export function getHandlerDataFor(namePage) {
  switch (namePage) {
    case "people":
      return getArrPeopleData;
    case "planets":
      return getArrPlanetsData;
    case "starships":
      return getArrStarshipsData;
    default:
      return getArrHomeworldData;
  }
}
