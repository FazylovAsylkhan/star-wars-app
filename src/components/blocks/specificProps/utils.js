function getAllExistSpecificProps(data) {
  return [
    { name: "Starships", values: data.starships },
    { name: "Vehicles", values: data.vehicles },
    { name: "Films", values: data.films },
    { name: "Species", values: data.species },
    { name: "Residents", values: data.residents },
    { name: "Pilots", values: data.pilots },
  ];
}

function filterProps(props) {
  const filteredProps = props.filter((prop) => {
    const { values } = prop;
    const IsExitValues = values !== undefined;
    const IsNotEmptyValues = values?.length > 0;

    return IsExitValues && IsNotEmptyValues ? prop : null;
  });

  return filteredProps;
}

const getFilteredProps = (data) => {
  let result = null;
  if (data) {
    const allExistSpecificProps = getAllExistSpecificProps(data);
    result = filterProps(allExistSpecificProps);
  }
  return result;
};

export default getFilteredProps;
