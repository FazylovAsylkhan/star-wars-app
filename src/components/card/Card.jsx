import React from "react";
import { Link } from "react-router-dom";
import s from "./card.module.scss";

function Card({ data, list, index }) {
  const { name, url } = data.results[index];
  const arr = url.split("/");
  const urlPersonalPage = arr[arr.length - 2];

  return (
    <>
      <br />
      {list && (
        <ul className={s.list}>
          {list.map(([props, value]) => (
            <li key={props}>
              <b>{props}: </b>
              {value}
            </li>
          ))}
          <Link to={urlPersonalPage}>{name}</Link>
        </ul>
      )}
    </>
  );
}

export default Card;
