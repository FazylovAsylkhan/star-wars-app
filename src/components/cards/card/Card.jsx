/* eslint-disable prefer-destructuring */
import React, { useState } from "react";
import ErrorBoundary from "../../errorBoundary/ErrorBoundary";
import InfoBox from "../../infoBox";
import ButtonMore from "../buttomMore";
import s from "./card.module.scss";

function Card({ results, index }) {
  const [list, setList] = useState(results);
  return (
    results && (
      <ErrorBoundary>
        <div data-testid="card" className={s.card}>
          <InfoBox results={list} />
          <ButtonMore idPersonalPage={index} />
          <button type="button" onClick={() => setList(null)}>
            Throw Error
          </button>
        </div>
      </ErrorBoundary>
    )
  );
}

export default Card;
