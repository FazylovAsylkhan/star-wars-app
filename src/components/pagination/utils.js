import s from "../button/button.module.scss";

function addActiveClass(url = null) {
  const currentNumberOfPage = url ? url.split("=")[1] : "1";
  const pagination = document.getElementById("pagination");
  const btns = Array.from(pagination.getElementsByTagName("button"));
  btns.forEach((btn) => {
    btn.classList.remove(s.activeBtn);
    if (btn.textContent === currentNumberOfPage) btn.classList.add(s.activeBtn);
  });
}

export default addActiveClass;
