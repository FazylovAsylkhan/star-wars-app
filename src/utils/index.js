export const transformSubDataToArray = (obj) =>
  obj.results.map((item) => Object.entries(item));

export const transformDataToArray = (item) => Object.entries(item);

export const getIdPageFrom = (list) => {
  let id = 0;
  if (list.length) {
    const url = list[list.length - 1][1];
    const arr = url.split("/");
    id = arr[arr.length - 2];
  }

  return id;
};
