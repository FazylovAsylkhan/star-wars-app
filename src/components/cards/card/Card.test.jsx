/* eslint-disable no-undef */
import React from "react";
import { screen } from "@testing-library/react";
import { renderWithRouter } from "../../../tests/testUtils";
import { transformDataToArray } from "../../../utils";
import testData from "../../../tests/testData";
import Card from "./Card";

describe("Card component works", () => {
  const handledData = transformDataToArray(testData.results[0]);
  it("should render Card component", async () => {
    renderWithRouter(<Card results={handledData} idPersonalPage={1} />);
    const card = screen.getByTestId("card");
    expect(card).toBeInTheDocument();
    expect(card).toMatchSnapshot();
  });
  it("Number of items in the card should be 15", () => {
    renderWithRouter(<Card results={handledData} idPersonalPage={1} />);
    const numberOfItems = screen.getAllByRole("listitem").length;
    expect(numberOfItems).toBe(8);
  });
});
