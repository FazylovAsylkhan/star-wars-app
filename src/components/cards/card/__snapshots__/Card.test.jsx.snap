// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component works should render Card component 1`] = `
<div
  class="card"
  data-testid="card"
>
  <ul
    class="list"
  >
    <li>
      <b>
        name
        : 
      </b>
      Luke Skywalker 1
    </li>
    <li>
      <b>
        birth_year
        : 
      </b>
      19BBY
    </li>
    <li>
      <b>
        created
        : 
      </b>
      2014-12-09T13:50:51.644000Z
    </li>
    <li>
      <b>
        edited
        : 
      </b>
      2014-12-20T21:17:56.891000Z
    </li>
    <li>
      <b>
        eye_color
        : 
      </b>
      blue
    </li>
    <li>
      <b>
        films
        : 
      </b>
      https://swapi.dev/api/films/1/
      https://swapi.dev/api/films/2/
      https://swapi.dev/api/films/3/
      https://swapi.dev/api/films/6/
    </li>
    <li>
      <b>
        gender
        : 
      </b>
      female
    </li>
    <li>
      <b>
        hair_color
        : 
      </b>
      blond
    </li>
  </ul>
  <button
    class="button primary"
    type="button"
  >
    <a
      class="link"
      href="/undefined"
    >
      More ...
    </a>
  </button>
  <button
    type="button"
  >
    Throw Error
  </button>
</div>
`;
